test/cases/regression-composer-works-behind-satellite.sh

```diff
@@ -0,0 +1,317 @@
+sslclientkey = ${PKI_DIR}/ca1/client.key
+sslclientcert = ${PKI_DIR}/ca1/client.crt
+metadata_expire = 86400
+enabled_metadata = 0
+
+[repo2]
+name = Repo 2 - local proxy
+baseurl = https://localhost:8009/repo
+enabled = 1
+gpgcheck = 0
+sslverify = 1
+sslcacert = ${PKI_DIR}/ca2/ca.crt
+sslclientkey = ${PKI_DIR}/ca2/client.key
+sslclientcert = ${PKI_DIR}/ca2/client.crt
+metadata_expire = 86400
+enabled_metadata = 0
+STOPHERE
+
+chattr +i ${REDHAT_REPO}
+lsattr ${REDHAT_REPO}
+cat ${REDHAT_REPO}
+
+# Allow httpd process to create network connections
+sudo setsebool httpd_can_network_connect on
+# Start httpd
+sudo systemctl start httpd || echo "Starting httpd failed"
+sudo systemctl status httpd
+
+greenprint "Verify dnf can use this configuration"
+sudo dnf install --repo=repo1 --repo=repo2 zsh -y
+
+greenprint "Rewrite osbuild-composer repository configuration"
+# In case this test case runs as part of multiple different test, try not to ruit the environment
+sudo mv "${REPOSITORY_OVERRIDE}" "${REPOSITORY_OVERRIDE}.backup"
+sudo tee "${REPOSITORY_OVERRIDE}" << STOPHERE
+{
+  "${ARCH}": [
+    {
+      "baseurl": "https://localhost:8008/repo",
+      "gpgkey": "${REPO1_GPGKEY}",
+      "check_gpg": false,
+      "rhsm": true
+    },
+    {
+      "baseurl": "https://localhost:8009/repo",
+      "gpgkey": "${REPO2_GPGKEY}",
+      "check_gpg": false,
+      "rhsm": true
+    }
+  ]
+}
+STOPHERE
+
+sudo systemctl restart osbuild-composer
+sudo composer-cli status show
+
+BLUEPRINT_FILE=/tmp/bp.toml
+BLUEPRINT_NAME=zishy
+
+cat > "$BLUEPRINT_FILE" << STOPHERE
+name = "${BLUEPRINT_NAME}"
+description = "A base system with zsh"
+version = "0.0.1"
+
+[[packages]]
+name = "zsh"
+STOPHERE
+
+COMPOSE_START=/tmp/compose-start.json
+COMPOSE_INFO=/tmp/compose-info.json
+sudo composer-cli blueprints push "$BLUEPRINT_FILE"
+if ! sudo composer-cli blueprints depsolve ${BLUEPRINT_NAME};
+then
+  sudo cat /var/log/httpd/error_log
+  sudo journalctl -xe --unit osbuild-composer
+  exit 1
+fi
+if ! sudo composer-cli --json compose start ${BLUEPRINT_NAME} qcow2 | tee "${COMPOSE_START}";
+then
+  sudo journalctl -xe --unit osbuild-composer
+  sudo journalctl -xe --unit osbuild-worker
+  exit 1
+fi
+COMPOSE_ID=$(jq -r '.build_id' "${COMPOSE_START}")
+
+# Wait for the compose to finish.
+greenprint "⏱ Waiting for compose to finish: ${COMPOSE_ID}"
+while true; do
+    sudo composer-cli --json compose info "${COMPOSE_ID}" | tee "${COMPOSE_INFO}" > /dev/null
+    COMPOSE_STATUS=$(jq -r '.queue_status' "$COMPOSE_INFO")
+
+    # Is the compose finished?
+    if [[ $COMPOSE_STATUS != RUNNING ]] && [[ $COMPOSE_STATUS != WAITING ]]; then
+        break
+    fi
+
+    # Wait 30 seconds and try again.
+    sleep 30
+
+done
+
+sudo journalctl -xe --unit osbuild-composer
+sudo journalctl -xe --unit osbuild-worker
+
+# Did the compose finish with success?
+if [[ $COMPOSE_STATUS != FINISHED ]]; then
+    echo "Something went wrong with the compose. 😢"
+    exit 1
+fi
```

Comment:
missing newline at the end of the file.

Reply:
`Fixing whitespaces, what else would I do ...`

Reply:
Do you consider this issue so pressing it needs a respin?

Reply:
No, I can live with it. Please just make sure that the issue is not present on `main`.